### PR TITLE
Remove deprecated ignore args

### LINKED ISSUE #2656: Remove deprecated argument in function iris.analysis.calculus.curl
The `ignore` argument in `iris.analysis.calculus.curl` was deprecated in Iris 0.8.

https://github.com/SciTools/iris/blob/master/lib/iris/analysis/calculus.py#L466

### LINKED ISSUE #2657: Remove deprecate `ignore` argument in iris.analysis.maths._add_subtract_common
The `ignore` argument in iris.analysis.maths._add_subtract_common was deprecated in Iris 0.8

https://github.com/SciTools/iris/blob/master/lib/iris/analysis/maths.py#L331

### PR DESCRIPTION
**Pull request by lbdreyer**
Fixes #2656 
Fixes #2657

### CODE REVIEW

**Comment by dkillick** [comment]
@lbdreyer you have not updated cube math functionality to match the changes you've made to the analysis/maths API; see [cube.py:L3080-L3109](https://github.com/SciTools/iris/blob/master/lib/iris/cube.py#L3080-L3109).

File: lib/iris/analysis/maths.py [hunk 1]
```diff
@@ -342,14 +340,6 @@ def _add_subtract_common(operation_function, operation_name, cube, other,
         # operation with
         coord_comp = iris.analysis.coord_comparison(cube, other)
 
-        # provide a deprecation warning if the ignore keyword has been set
-        if ignore is not True:
-            msg = ('The "ignore" keyword has been deprecated in '
-                   'add/subtract. This functionality is now automatic. '
-                   'The provided value to "ignore" has been ignored, '
-                   'and has been automatically calculated.')
-            warn_deprecated(msg)
```
Comment (dkillick) [inline lib/iris/analysis/maths.py]: Now that you've removed this usage of `warn_deprecated` you'll also need to remove the import on [L34](https://github.com/lbdreyer/iris/blob/755fb3da11633443a86e1d1375004f1b36b0741d/lib/iris/analysis/maths.py#L34).

**Comment by lbdreyer** [comment]
> you have not updated cube math functionality to match the changes you've made to the analysis/maths API

I had tried grep-ing for usages of add and subtract  but clearly didn't find these

**Comment by dkillick** [comment]
@lbdreyer don't worry! I only knew they were there because I was doing some work in that area as part of the dask integration work...

**Comment by corinnebosley** [comment]
@dkillick I am happy with Laura's changes here, and will merge if you can confirm that the review changes that you requested are now approved. (I have checked them but I don't like to see a big red cross sitting on a PR that I want to merge.)

**Comment by corinnebosley** [comment]
Build failed due to Travis Python version issue: https://github.com/SciTools/iris/pull/2769#issuecomment-335147830

Issue has been resolved, but I am unsure of next steps...

**Pull request merged**